service/history/execution/mutable_state_builder_methods_activity.go

```diff
@@ -290,7 +290,7 @@ func (e *mutableStateBuilder) tryDispatchActivityTask(
 		metrics.WorkflowTypeTag(e.GetWorkflowType().Name),
 		metrics.TaskListTag(ai.TaskList))
 	taggedScope.IncCounter(metrics.DecisionTypeScheduleActivityDispatchCounter)
-	err := e.shard.GetService().GetMatchingClient().AddActivityTask(ctx, &types.AddActivityTaskRequest{
+	_, err := e.shard.GetService().GetMatchingClient().AddActivityTask(ctx, &types.AddActivityTaskRequest{
```

Comment:
At which layer we will be caching these results and make use of it? If it's going to be done by a matching client wrapper then we probably don't need to change `matching.Client` interface and the callers wouldn't change. I may be missing something (later review)

Reply:
it's done in `matching.Client`. Since I've made the change, let's keep it. It's always better to have an empty response than void

Reply:
Discussed offline. We may actually need this response to record the identity of Started tasks for a feature that reduces amount of transfer tasks.